### DISCUSSION TITLE
tslab vs itypescript re: redefining variables and function

### DISCUSSION THREAD

**Discussion by MikeMcDonald83** [yunabe/tslab#81] [General]
In this [SO answer](https://stackoverflow.com/questions/55080181/typescript-in-jupyter-notebook), you wrote,

> Even if you could install itypescript kernel correctly, you may find the kernel is not very useful in your use case because it does not allow you to redefine a variable with the same name by design. If you execute function f() {}, you just get an error message like TS2393: Duplicate function implementation..

But in your kernel, when I have two functions that are named the same in different cells, I get the "duplicate function implementation" error. The same is true for scoped variables. Am I misunderstanding what you meant by that answer? Is the TS linter just confused about what's possible? Should I be able to redefine functions and scoped variables in your kernel? If not, how is that different than itypescript? Can you please explain what distinction you were making in that answer? Thanks!

**Comment by yunabe** [comment]
> I get the "duplicate function implementation" error.

That's strange. This kernel is designed to overwrite the new functions too (See the screenshot below). If the kernel fails to overwrite identifiers defined in previous cells, that is a bug.

<img width="760" alt="screenshot" src="https://user-images.githubusercontent.com/255937/234615472-c5c0b489-cd20-4f23-9b46-fbbc0dc1d353.png">

I'm not sure how "TS linter" is related in this context.
